Source/JavaScriptCore/bytecode/ObjectPropertyConditionSet.h

```diff
@@ -168,7 +168,7 @@ ObjectPropertyConditionSet generateConditionsForPropertySetterMiss(
 ObjectPropertyConditionSet generateConditionsForIndexedMiss(VM&, JSCell* owner, JSGlobalObject*, Structure* headStructure);
 ObjectPropertyConditionSet generateConditionsForPrototypePropertyHit(
     VM&, JSCell* owner, JSGlobalObject*, Structure* headStructure, JSObject* prototype,
-    UniquedStringImpl* uid);
+    UniquedStringImpl* uid, bool prototypeEquivalenceConditionKind = false);
```

Comment:
Let's not add default parameter with boolean. It is too hard to imagine what it means.

1. Not using default parameter. Always specify it explicitly.
2. We should instead passing PropertyCondition.

Reply:
Thank you Yusuke, I was contemplating between `PropertyCondition::Kind` that only accepts 2 values of 8 (a bit weird) vs a boolean flag, and slightly leaned to the latter.

Changed to accept `PropertyCondition::Kind` + asserting the kind is sane.